freqtrade/strategy/interface.py

```diff
@@ -1379,6 +1383,8 @@ def advise_entry(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
         logger.debug(f"Populating enter signals for pair {metadata.get('pair')}.")
 
         df = self.populate_entry_trend(dataframe, metadata)
+        if self.config.get('runmode') not in [RunMode.DRY_RUN, RunMode.LIVE]:
+            optimize_dtypes(dataframe)
```

Comment:
I think this will have an adverse effect than what you intend.

I can imagine calling this after `populate_indicators()` - but if you call it after "entry_trend" AND `exit_trend` - you'll call this twice for every hyperopt iteration (within each epoch), after an initial call before the hyperopt epochs (that's the one from `populate_indicators` - which i can see making sense).


As each iteration does operations on every column (therefore, for indicator columns, performing it a total of 3 times without actually changing anything) - it'll have an adverse affect, causing a lot more work than what would actually be necessary, thereby negating the slight benefit a different type MAY have.

Reply:
so to be clear, use are suggesting removing the function call from `advise_entry` and `advise_exit`; right?

Reply:
yes (unless you have compelling tests to show otherwise).

Doing such optimization is always a tradeoff - you're trading higher RAM for longer CPU time (more operations to reduce ram).

up to some point, the tradeoff may be in favor of "do more CPU" (so do the conversation) - but in this case, i think your overall gain (more speed for hyperopt) - the RAM decrease will probably not be worth it, as you'll trade in "more CPU" once every epoch - which is 100 or 1000'ds of times (depending on -e in hyperopt).

Reply:
Is there any benchmarking guideline for freqtrade so we can decide on this tradeoff better?
if there isn't, how can I showcase credible results for these gains I'm mentioning? What was your previous approach to this kind of PRs?